### ISSUE TITLE
[bug] blocked subdomain doesn't display correctly in UI

### ISSUE THREAD

**Issue by coxde** [superseriousbusiness/gotosocial#3089]
### Describe the bug with a clear and concise description of what the bug is. Please include screenshots of any visual issues.

Hiii again! When I block a domain `example.org` in the Admin UI, GoToSocial should block all subdomains such as `sub.example.org`. However, in Accounts Search page, it doesn't display strikethrough for `@account@sub.example.org` or `Suspended: Yes` in the Account Details page. But by checking the log and trying to search the account, the instance should be already blocked (I think?)

### What's your GoToSocial Version?

v0.16.0+git-f1cbf6f

### Browser version

Firefox 127.0.2

### How to reproduce it?

1. Block `example.org` in Domain Blocks page
2. Check `sub.example.org` in Accounts Search page
3. Find out that it does not appear to be blocked according to the UI

**Comment by tsmethurst** [comment]
Hiya, yeah the domain is definitely blocked, don't worry. Every HTTP request coming into your instance is checked against domain blocks, and that includes subdomains, and any requests that match the domain or a subdomain will be denied with 403. So if you've blocked `example.org`, a request coming from `example.org` is blocked, and any requests coming from subdomains will be blocked too (`sub.example.org`, `deep.sub.example.org`).

The strikethrough indicates that the account is marked "suspended" in the database. Currently when a domain block is first created, we select all accounts in the database that have that exact domain, and mark them as suspended. So when you block `example.org`, all accounts with domain `example.org` are marked suspended as a side effect. This ensures that all their data and statuses and media and stuff are also removed, since it's unlikely you'll want to keep them around.

However, we need to rework that select to also include subdomains, to tidy their statuses and media up too. See https://github.com/superseriousbusiness/gotosocial/issues/2146

So to summarize: it's blocked, all data from accounts on the domain will be removed when they're marked suspended; the subdomain is blocked too, but currently we don't have a neat way of also removing data from subdomains.

**Comment by coxde** [comment]
Ohh thank you! Sorry I didn't notice https://github.com/superseriousbusiness/gotosocial/issues/2146, I guess I can close this issue then.

**Comment by tsmethurst** [comment]
No worries :)
